bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxWashingMachineHandler.java

```diff
@@ -0,0 +1,173 @@
+                        .getProperties().getReported().getFCMiscellaneousState().getDetergentExtradosage()));
+            case SOFTENER_EXTRA_DOSAGE:
+                return new StringType(Integer.toString(((WashingMachineStateDTO) dto.getApplianceState())
+                        .getProperties().getReported().getFCMiscellaneousState().getSoftenerExtradosage()));
+            case WATER_USAGE:
+                return new StringType(Integer.toString(((WashingMachineStateDTO) dto.getApplianceState())
+                        .getProperties().getReported().getFCMiscellaneousState().getWaterUsage()));
+            case TOTAL_WASH_CYCLES_COUNT:
+                return new StringType(Integer.toString(((WashingMachineStateDTO) dto.getApplianceState())
+                        .getProperties().getReported().getTotalWashCyclesCount()));
+            case CYCLE_PHASE:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getCyclePhase());
+            case APPLIANCE_TOTAL_WORKING_TIME:
+                return new StringType(Integer.toString(((WashingMachineStateDTO) dto.getApplianceState())
+                        .getProperties().getReported().getApplianceTotalWorkingTime()));
+            case APPLIANCE_STATE:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getApplianceState());
+            case APPLIANCE_MODE:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getApplianceMode());
+            case ANALOG_TEMPERATURE:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getUserSelections().getAnalogTemperature());
+            case ANALOG_SPIN_SPEED:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getUserSelections().getAnalogSpinSpeed());
+            case STEAM_VALUE:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getUserSelections().getSteamValue());
+            case PROGRAMS_ORDER:
+                return new StringType(((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported()
+                        .getUserSelections().getProgramUID());
+
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    public Map<String, String> refreshProperties() {
+        Map<String, String> properties = new HashMap<>();
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ElectroluxAppliancesBridgeHandler bridgeHandler = (ElectroluxAppliancesBridgeHandler) bridge.getHandler();
+            if (bridgeHandler != null) {
```

Comment:
Another `instanceof` pattern matching improvement can be made here.

Reply:
Fixed!